pkg/app/zipper/tldcache.go

```diff
@@ -34,12 +45,64 @@ func (app *App) probeTopLevelDomains(ms *PrometheusMetrics) {
 	}
 }
 
+type tldPrefix struct {
+	prefix  string
+	nsCount int
+}
+
+func sortedByNsCount(prefixes []string) []string {
+	countedPrefixes := make([]tldPrefix, len(prefixes))
+	for i, prefix := range prefixes {
+		nsCount := 0
+		if prefix != "" {
+			nsCount = strings.Count(prefix, ".") + 1
+		}
+		countedPrefixes[i] = tldPrefix{
+			prefix:  prefix,
+			nsCount: nsCount,
+		}
+	}
+	sort.Slice(countedPrefixes, func(i, j int) bool {
+		return countedPrefixes[i].nsCount < countedPrefixes[j].nsCount
+	})
+	sortedPrefixes := make([]string, len(prefixes))
+	for i := range countedPrefixes {
+		sortedPrefixes[i] = countedPrefixes[i].prefix
+	}
+	return sortedPrefixes
+}
+
+// getBackendsForPrefix returns the backends that need to be queried in order to populate TLD cache for the prefix.
+// it reuses already fetched tlds to find out about the info. if no info is there, it returns all the backends.
```

Comment:
nit: if it's formatted as proper sentences w/ dots, then might be a good idea to also start the sentences from capital letter

Reply:
It's the name of the function. Capitalizing it, make it wrong.

Reply:
I meant "It reuses .." and "If no info is there .." sentences.

Reply:
Oh sorry. Fixed.